src/components/AuthorViewpoint/FactListCard.tsx

```diff
@@ -0,0 +1,117 @@
+"use client";
+import { Fact } from "@/types/conversations.types";
+import EditViewpointFact from "@/components/AuthorViewpoint/EditViewpointFact";
+import { MagnifyingGlassIcon, PlusIcon } from "@heroicons/react/24/outline";
+import { Select, Button } from "@mantine/core";
+import { useState, useEffect, Dispatch, SetStateAction } from "react";
+import { allFacts } from "@/mock/conversationMock";
+
+type FactListCardProps = {
+    viewpointFactList: Fact[];
+    setViewpointFactList: Dispatch<SetStateAction<Fact[]>>;
+};
+
+export default function FactListCard({
+    viewpointFactList,
+    setViewpointFactList,
+}: FactListCardProps) {
+    const [searchData, setSearchData] = useState<Fact[]>(allFacts);
+    const [selectedFactId, setSelectedFactId] = useState<string>("");
+    const [searchValue, setSearchValue] = useState<string>(""); // eslint-disable-line
+
+    useEffect(() => {
+        const selectedFact = searchData.find(
+            (fact) => String(fact.id) === selectedFactId,
+        );
+        if (selectedFact) {
+            //move the selected fact to the viewpointFactList
+            setViewpointFactList((prev) => [...prev, selectedFact]);
+
+            //remove the selected fact from the searchData
+            const newSearchData = searchData.filter(
+                (fact) => fact.id !== selectedFact.id,
+            );
+            setSearchData(newSearchData);
+            setSelectedFactId("");
+        }
```

Comment:
The logic here should be separated into checking whether a fact is selected, and if yes, then whether the id is findable. If no match for a selected ID in the searchData is found (which shouldn't be possible), an error should be thrown to help us locate the bug that's causing this.